test/adapters/adapters.spec.js

```diff
@@ -0,0 +1,199 @@
+    logger.log(`juttle -e "${juttle}"`);
+    return docker.exec('juttle-engine-local',
+                       ['/opt/juttle-engine/bin/juttle',
+                        '-e',
+                        juttle], { quiet: true });
+}
+
+function waitForSuccess(containerName, args) {
+    return retry(() => {
+        return docker.exec(containerName, args, { quiet: true });
+    }, {
+        max_tries: 10,
+        interval: 1000
+    });
+}
+
+describe('adapters', () => {
+
+    before(() => {
+        var tmpdir = tmp.dirSync().name;
+        return Promise.all([
+            docker.destroy('elasticsearch-adapter-test'),
+            docker.destroy('influxdb-adapter-test'),
+            docker.destroy('graphite-adapter-test'),
+            docker.destroy('juttle-engine-local')
+        ])
+        .then(() => {
+            return docker.checkJuttleEngineLocalExists();
+        })
+        .then(() => {
+            return Promise.all([
+                docker.run({
+                    name: 'elasticsearch-adapter-test',
+                    image: 'elasticsearch:1.5.2',
+                    detach: true
+                }),
+                docker.run({
+                    name: 'influxdb-adapter-test',
+                    image: 'tutum/influxdb:0.10',
+                    detach: true
+                }),
+                docker.run({
+                    name: 'graphite-adapter-test',
+                    image: 'sitespeedio/graphite:0.9.14',
+                    detach: true
+                })
+            ]);
+        })
+        .then(() => {
+            var config = JSON.stringify({
+                adapters: {
+                    elastic: {
+                        address: 'elasticsearch',
+                        port: 9200
+                    },
+                    influx: {
+                        url: 'http://influxdb:8086'
+                    },
+                    graphite: {
+                        carbon: {
+                            host: 'graphite',
+                            port: 2003
+                        },
+                        webapp: {
+                            host: 'graphite',
+                            port: 80,
+                            username: 'guest',
+                            password: 'guest'
+                        }
+                    }
+                }
+            });
+
+            var filename = path.join(tmpdir, '.juttle-config.json')
+            return fs.writeFileAsync(filename, config)
+        })
+        .then(() => {
+            return docker.run({
+                name: 'juttle-engine-local',
+                image: 'juttle-engine:local',
+                ports: ['8080:8080'],
+                links: [
+                    'elasticsearch-adapter-test:elasticsearch',
+                    'influxdb-adapter-test:influxdb',
+                    'graphite-adapter-test:graphite'
+                ],
+                volumes: [`${tmpdir}:/tmp`],
```

Comment:
If you agree with the above comment, you'd have to map the config file in differently as well.